### PR TITLE
Tests: Warning clean up 

### PR DESCRIPTION
**Pull request by daschuer**
I did this PR during hunting down this Launchpad PPA build failure: 
https://launchpadlibrarian.net/482151056/buildlog_ubuntu-eoan-i386.mixxx_2.3.0~beta~git7423-0ubuntu1~2.3~git7423~eoan_BUILDING.txt.gz

The failing test is already fixed now, unsure why. Maybe due to https://github.com/mixxxdj/mixxx/pull/2792

### CODE REVIEW

File: src/mixer/basetrackplayer.cpp [hunk 1]
```diff
@@ -180,6 +180,9 @@ BaseTrackPlayerImpl::BaseTrackPlayerImpl(QObject* pParent,
     m_pPlay = std::make_unique<ControlProxy>(group, "play", this);
     m_pPlay->connectValueChanged(this, &BaseTrackPlayerImpl::slotPlayToggled);
 
+    m_pRateRatio = std::make_unique<ControlProxy>(group, "rate_ratio", this);
```
Comment (uklotzde) [inline src/mixer/basetrackplayer.cpp:183]: Why unique_ptr instead of parented_ptr?

File: src/mixer/basetrackplayer.cpp [hunk 2]
```diff
@@ -440,22 +443,22 @@ void BaseTrackPlayerImpl::slotTrackLoaded(TrackPointer pNewTrack,
 
         if(m_pConfig->getValue(
                 ConfigKey("[Mixer Profile]", "EqAutoReset"), false)) {
-            if (m_pLowFilter != NULL) {
+            if (m_pLowFilter != nullptr) {
```
Comment (uklotzde) [inline src/mixer/basetrackplayer.cpp:446]: Omit the ` != nullptr` entirely

**Comment by daschuer** [comment]
Done

**Comment by uklotzde** [comment]
LGTM, waiting for CI.

**Pull request merged**